### ISSUE TITLE
False positive inside list comprehension

### ISSUE THREAD

**Issue by last-partizan** [microsoft/pyright#5690]
**Describe the bug**
Just take a look at the example, when using just `round()` it works fine, but the same call inside list comprehension is marked as error.

**Code or Screenshots**

```python
from __future__ import annotations

from decimal import Decimal

v: Decimal | int = 1

round(v, 2) # Fine

rounded = [
    round(v, 2)  # error: Argument of type "Literal[2]" cannot be assigned to parameter "ndigits" of type "None"
    for v in [Decimal("10.000"), 1]
]
```

**VS Code extension or command-line**

* pyright 1.1.321

**Comment by erictraut** [comment]
Thanks for the bug report. Although it's not immediately obvious, this is caused by the same underlying problem as https://github.com/microsoft/pyright/issues/5673. This bug has been fixed and will be addressed in the next release.

**Comment by erictraut** [comment]
This regression has affected several pyright users in various ways, so I decided to do a hot-fix release. I just published 1.1.322, which addresses the issue.
